src/main/java/frc/robot/subsystems/swerveSubsystem/SwerveSubsystem.java

```diff
@@ -184,13 +193,58 @@ public double getMagnitude() {
     }
   }
 
-  public record DriveInput(double x, double y, double alpha) {}
-
   @Override
   public void periodic() {
     if (this.getCurrentCommand() != null) currentCommand.log(this.getCurrentCommand().getName());
   }
 
+  private Rotation2d getOperatorPerspective() {
+    return m_operatorForwardDirection;
+  }
+
+  public static class RotationSource implements DoubleSupplier {
+    private double lastVal;
+    private final Supplier<Translation2d> supplier;
+    private final SwerveSubsystem subsystem;
+
+    public RotationSource(XboxController controller, SwerveSubsystem subsystem) {
+      this.subsystem = subsystem;
+      supplier = () -> new Translation2d(controller.getRightX(), controller.getRightY());
+      lastVal = subsystem.getState().Pose.getRotation().getDegrees();
+    }
+
+    @Override
+    public double getAsDouble() {
+      Translation2d input =
+          supplier
+              .get()
+              .rotateBy(
+                  DriverConstants.ABSOLUTE_POINTING_OFFSET.plus(
+                      subsystem.getOperatorPerspective()));
+      SmartDashboard.putNumber("last angle", lastVal);
```

Comment:
Is this needed for competition?  If so, chose a tab for it.  If not, let's put it on an isCompetition block or debugEntry